### ISSUE TITLE
Replace the "OPTIONAL" entity qualifier with "CONDITIONAL" where appropriate

### ISSUE THREAD

**Issue by Moo-Marc** [bids-standard/bids-specification#563]
As was discussed while clarifying that session labels are REQUIRED in some cases (https://github.com/bids-standard/bids-specification/pull/532), thus not exactly OPTIONAL as defined in rfc2119 ("truly optional"), it appears this is the case for many other (most?) OPTIONAL entities in the entities table.

We could therefore mark them as being CONDITIONAL instead.

While this table is being modified to be machine readable (https://github.com/bids-standard/bids-specification/pull/475), I think it would be ideal to add a link to the section where the specific conditions are described for each entity, if possible.

**Comment by sappelhoff** [comment]
I agree that it'd be nice to have a solution for the "conditionally optional" as opposed to "truly optional" cases.

Personally, I think it could be a nice idea to extend rfc2119 to include one more definition for that reason (the CONDITIONAL you propose).

However, @effigies recently uttered a contrary opinion.
